xbmc/addons/AddonDll.cpp

```diff
@@ -46,6 +48,7 @@ CAddonDll::CAddonDll(AddonProps props)
   m_pHelpers    = NULL;
   m_needsavedsettings = false;
   m_parentLib.clear();
+  memset(&m_interface, 0, sizeof(m_interface));
```

Comment:
this is generally not a good way to initialize a struct/class, it'll destroy the vtable if it happens to have virtual methods.